WebApi/WebApi/Models/DTO/PostDTOs/PostDTO.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace WebApi.Models.DTO
+{
+    public class PostDTO : IValidatableObject
+    {
+        [Required]
+        public int id { get; set; }
+        [Required]
+        [MaxLength(50)]
+        public string title { get; set; }
+        [Required]
+        public string content { get; set; }
+        [Required]
+        public DateTime datetime { get; set; }
+        [Required]
+        public string category { get; set; }
+        [Required]
+        public bool isPromoted { get; set; }
+        [Required]
+        public string author { get; set; }
+        [Required]
+        public int authorID { get; set; }
+        [Required]
+        public int likesCount { get; set; }
+        [Required]
+        public bool isLikedByUser { get; set; }
+
+
+        //[Required]
+        //public IQueryable<CommentDTO> comments { get; set; }
+        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+        {
```

Comment:
tutaj jeszcze musi być walidacja DateTime - jest to struktura, jeżeli jej nie podamy w body json to zostanie podstawiona domyślna wartość (jakiś tam pierwszy dzień) - DateTime().
Jeżeli wartość tego pola to DateTime() to nie przepuszczamy!